e2e/connection_test.go

```diff
@@ -150,11 +158,38 @@ func consumerConnectFlow(t *testing.T, tequilapi *tequilapi_client.Client, consu
 	assert.NoError(t, err)
 
 	// sessions history should be updated after disconnect
-	sessionsDTO, err = tequilapi.GetSessions()
+	sessionsDTO, err = tequilapi.GetSessionsByType(serviceType)
 	assert.NoError(t, err)
+
 	assert.Equal(t, 1, len(sessionsDTO.Sessions))
 	se = sessionsDTO.Sessions[0]
-	assert.NotEqual(t, uint64(0), se.BytesSent)
-	assert.NotEqual(t, uint64(0), se.BytesReceived)
-	assert.Equal(t, "Completed", se.Status)
+
+	// call the custom asserter for the given service type
+	serviceTypeAssertionMap[serviceType](t, se)
+}
+
+type sessionAsserter func(t *testing.T, session endpoints.SessionDTO)
+
+var serviceTypeAssertionMap = map[string]sessionAsserter{
+	"openvpn":   assertOpenvpn,
+	"noop":      assertNoop,
+	"wireguard": assertWireguard,
+}
+
+func assertOpenvpn(t *testing.T, session endpoints.SessionDTO) {
```

Comment:
What's the point of different service type assertions ? Function is the same :|

Reply:
fixed, now we're asserting `assertStatsNotZero` and `assertStatsZero